### ISSUE TITLE
Failure in `rotation_test_cft`

### ISSUE THREAD

**Issue by eddyashton** [microsoft/CCF#2847]
The `rotation_test_cft` has started failing in the CI with a very odd error:

https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=29814&view=logs&j=8f3dc89c-3708-5926-47e7-27120a268dab&t=d37f6062-24ee-5eff-15b2-0eb74651c675&l=24705

```
67:   File "/mnt/build/2/s/tests/infra/checker.py", line 46, in check_can_progress
67:     c.wait_for_commit(r, timeout=timeout)
67:   File "/mnt/build/2/s/python/ccf/clients.py", line 672, in wait_for_commit
67:     raise ValueError("Response seqno and view should not be None")
67: ValueError: Response seqno and view should not be None
60/68 Test #67: rotation_test_cft .................***Failed   16.69 sec
```

```
67: 09:53:26.409 | INFO     | infra.network:wait_for_new_primary:891 - Waiting up to 16.0s for a new primary different from 0 (ccd4d5694d9a48b86ec88b8fec8c5dc5a146ce459220fe58713321caa264b8f3) to be elected...
67: 09:53:29.349 | INFO     | infra.network:wait_for_new_primary:918 - [1||] GET /node/network
67: 09:53:29.349 | INFO     | infra.network:wait_for_new_primary:918 - 200 @2.21 {"current_view":3,"primary_id":"af2b20a052f5240fc545d9259973c5ad7822a262d291fd7b65581701e5512812","service_certificate":"-----BEGIN CERTIFICATE-----\nMIIBtjCCAT2gAwIBAgIQZHhrgRuppbrh0mswKDkQdjAKBggqhkjOPQQDAzAWMRQw\nEgYDVQQDDAtDQ0YgTmV0d29yazAeFw0yMTAzMTEw + 559 chars
67: 09:53:29.349 | INFO     | infra.network:wait_for_new_primary:919 - New primary is 1 (af2b20a052f5240fc545d9259973c5ad7822a262d291fd7b65581701e5512812) in term 3
67: 09:53:29.529 | INFO     | infra.checker:check_can_progress:44 - [2|user0|] POST /app/log/private {'id': 42, 'msg': 'Hello world'}
67: 09:53:29.529 | INFO     | infra.checker:check_can_progress:44 - 404 {"error":{"code":"ResourceNotFound","message":"Unknown path: /log/127.211.183.156:45423."}}
```

I can't work out where `/log/127...` comes from, it looks like we're mangling the HTTP request line somewhere.

**Comment by eddyashton** [comment]
Also seen here: https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=29661&view=logs&j=ee9484ec-b476-577b-aa0e-d5ddfcbcd149&s=d654deb9-056d-50a2-1717-90c08683d50a&t=8e7d12cf-69d2-5704-84b7-b6fd04033d7b&l=22778

So this has happened in both NonSGX Debug and SGX Debug, though is inconsistent in both.

**Comment by eddyashton** [comment]
I've got more verbose logging from #2848, but still have no idea where this is getting mangled. It appears to be mangled by `requests`, before it ever reaches the node's HTTP parsing.

https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=29836&view=logs&j=8f3dc89c-3708-5926-47e7-27120a268dab&t=d37f6062-24ee-5eff-15b2-0eb74651c675&l=992

```
67: 10:35:37.930 | DEBUG    | ccf.clients:request:451 - <cyan>POST</> <green>/app/log/private</> {'id': 42, 'msg': 'Hello world'}
67: 10:35:37.930 | DEBUG    | ccf.clients:request:452 - https://127.19.30.9:42073/app/log/private
67: 10:35:37.931 | DEBUG    | ccf.clients:request:453 - b'{"id": 42, "msg": "Hello world"}'
67: 10:35:38.066 | INFO     | infra.checker:check_can_progress:44 - [2|user0|] POST /app/log/private {'id': 42, 'msg': 'Hello world'}
67: 10:35:38.066 | INFO     | infra.checker:check_can_progress:44 - 404 {"error":{"code":"ResourceNotFound","message":"Unknown path: /log/127.115.111.128:45581."}}
```

Those `DEBUG` lines list what we then pass to requests:
```python
            if request.verbose:
                LOG.debug(f"{request}")
                LOG.debug(f"https://{self.host}:{self.port}{request.path}")
                LOG.debug(f"{request_body}")
            response = self.session.request(
                request.http_verb,
                url=f"https://{self.host}:{self.port}{request.path}",
                auth=auth_value,
                headers=extra_headers,
                allow_redirects=request.allow_redirects,
                timeout=timeout,
                data=request_body,
            )
```

And then on Node 2's log:
```
2021-07-27T10:35:38.066099Z -0.007 0   [info ] ../src/http/http_endpoint.h:66       | !!! Going to parse 230 bytes:
POST /app/log/127.115.111.128:45581 HTTP/1.1
Host: 127.19.30.9:42073
...
```

I guess we must be going through the redirect path, and confusing requests with our redirect headers...

**Comment by eddyashton** [comment]
Resolved in #2849

**Comment by achamayou** [comment]
Using wait_for_new_primary() without specifying nodes is never safe if there is a reconfiguration.
